sdks/java/core/src/main/java/org/apache/beam/sdk/options/CustomPipelineOptionsInitializer.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.beam.sdk.options;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Iterators;
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+
+/**
+ * Interface to support offspring wire-in for Li: if input class meets some requirements,
+ * the customized pipelineOptions will be initialized.
+ *
+ */
+@SuppressWarnings("rawtypes")
+public interface CustomPipelineOptionsInitializer<T> {
+  void init(T pipelineOptions, Class<T> clazz);
+
+  interface Registrar {
```

Comment:
how will the Registrar be used?

Reply:
In the implementation, we will have a class annotated as AutoService(CustomePipelineOptionsInitializer.Registrar.class). It returns an instance of the implementation, so beam will know which implementation to use during runtime

Reply:
Got it. Can you add an example of usage in the javadoc?

Reply:
Updated